website/content/docs/api-gateway/upgrade-specific-versions.mdx

```diff
@@ -0,0 +1,283 @@
+- **Optional** [jq](https://stedolan.github.io/jq/download/) is installed on the users CLI
+
+
+
+### Pre-requisites
+
+In order to follow this guide, the following conditions must be true:
+
+- You have the ability to run Kubectl CLI commands
+- Your kubectl config is already configured to point at the cluster with the installation you are upgrading
+- You have HTTPRoute.read, TCPRoute.read and ReferencePolicy.create rights on your Kubernetes cluster
+
+
+### Procedures
+
+**1.** Optional, double check the current version of the consul-api-gateway with the following command:
+
+```shell-session
+$ kubectl get deployment --namespace consul consul-api-gateway-controller --output=jsonpath= "{@.spec.template.spec.containers[?(@.name=='api-gateway-controller')].image}"
+```
+
+If the output looks as follows:
+
+```log
+"hashicorp/consul-api-gateway:0.1.0"
+```
+
+ Continue following this upgrade path.
+
+**2.** Retrieve all routes that have a backend in a different namespace:
+
+There are two ways to retrieve cross-namespace routes:
+
+1. Method 1 is more manual but requires no additional dependencies
+1. Method 2 is faster but requires that [jq](https://stedolan.github.io/jq/) be installed
+
+##### Method 1: Manual
+
+Get all HTTPRoutes and TCPRoutes across all namespaces with the following command:
+
+```shell-session
+$ kubectl get HTTPRoute,TCPRoute -o json -A
+```
+
+If you have any active HTTPRoutes or TCPRoutes, you should receive output that looks as follows. Note that the output has been truncated to show only relevant fields.
+
+Note, the above command will retrieve all HTTPRoutes and TCPRoutes. TLSRoutes and UDPRoutes are not supported in v0.1.0.
+
+```yaml
+...
+
+apiVersion: v1
+items:
+- apiVersion: gateway.networking.k8s.io/v1alpha2
+  kind: HTTPRoute
+  metadata:
+    name: example-http-route,
+    namespace: example-namespace,
+    ...
+  spec:
+    parentRefs:
+    - group: gateway.networking.k8s.io
+      kind: Gateway
+      name: gateway
+      namespace: gw-ns
+    rules:
+    - backendRefs:
+      - group: ""
+        kind: Service
+        name: web-backend
+        namespace: gateway-namespace
+        ...
+      ...
+- apiVersion: gateway.networking.k8s.io/v1alpha2
+  kind: TCPRoute
+  metadata:
+    name: example-tcp-route,
+    namespace: a-different-namespace,
+    ...
+  spec:
+    parentRefs:
+    - group: gateway.networking.k8s.io
+      kind: Gateway
+      name: gateway
+      namespace: gateway-namespace
+    rules:
+    - backendRefs:
+      - group: ""
+        kind: Service
+        name: web-backend
+        namespace: gateway-namespace
+        ...
+...
+
+```
+
+For each of the above defined routes, you will need to inspect each of the backendRefs.
+
+If a backendRef has no namespace field defined or the namespace matches the namespace of the route itself, that backendRef requires no further action.
```

Comment:
I'm curious in general if we should be using inline code blocks (so `backendRef` instead of backendRef) where we mention things like backendRef, ReferencePolicy, Deployment, Service, HTTPRoute, TCPRoute, etc. 

I didn't see any clear guidance on this in the [Markdown](https://github.com/hashicorp/engineering-docs/blob/main/writing/markdown.md#inline-styles) or [Writing](https://github.com/hashicorp/engineering-docs/blob/main/writing/style-guide.md#text-formatting) style guides.

Could you speak to this @Jeff-Apple?

Reply:
@nathancoleman , When referring to a config parameter/object/etc. that the user is dealing with, we usually use prefer inline code blocks. Here are two examples of when I would and wouldn't use inline code blocks:

"When you configure a `HTTPRoute`, you will..."
"When you configure a route, you will..."

For "dictionary" words such as gateway, that are also config parameters, it depends on if the word is being used in a generic way or referring to the config parameter. Here is how I would write the sentence from line 129:

"If a `backendRef` does not have a `namespace` field defined or the namespace matches the namespace of the route itself, that `backendRef` requires no further action."

Reply:
Once we're able to manage docs in our own repo/versioning it may be nice to add a lint task to check for improper capitalization of these terms.